store.go

```diff
@@ -1641,13 +1641,11 @@ func (s *store) CreateImage(id string, names []string, layer, metadata string, i
 				options.Metadata = iOptions.Metadata
 			}
 			options.BigData = append(options.BigData, copyImageBigDataOptionSlice(iOptions.BigData)...)
-			options.NamesHistory = append(options.NamesHistory, copyStringSlice(iOptions.NamesHistory)...)
+			options.NamesHistory = append(options.NamesHistory, iOptions.NamesHistory...)
```

Comment:
I only skimmed/spot checked but I noticed in this one we stopped copying...expected?